components/AddProject.js

```diff
@@ -0,0 +1,63 @@
+import styled from "styled-components";
+export default function AddProject({ addProject }) {
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    // NEW
+    const newData = {
+      ...data,
+      slug: data.title,
+      id: crypto.randomUUID(),
+      tasks: [],
+    };
+
+    addProject(newData);
+
+    event.target.reset();
+  }
+
```

Comment:
the way you wrote the functions, it is unclear from outside the component what steps, what logic is applied to convert the form submission into data. I even tend to handle the event logic in the parent component to keep the component as dumb as possible.

Reply:
Thanks for the review 😊 I tried moving it to the parent component, but it is not working out like I wanted to. I really would like to possibly leave it like this, just because I want to avoid my app crashing again.